### PR TITLE
Concolic exec and tests

### LINKED ISSUE #442: Implement concolic analysis for laser

### PR DESCRIPTION
**Pull request by JoranHonig**
This pr includes a bit of logic to make concolic execution easy, this is ultimately to help the integration of evm testsuite. 
Which this pr also does. The initial framework has been setup and all the addition tests have been added

closes #442

### CODE REVIEW

File: mythril/laser/ethereum/transaction/transaction_models.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def initial_global_state(self):
             self.gas_price,
             self.call_value,
             self.origin,
-            code=self.callee_account.code if self.code is None else self.code,
+            code=self.code or self.callee_account.code,
```
Comment (norhh) [inline mythril/laser/ethereum/transaction/transaction_models.py]: perfect :1st_place_medal: . I forgot such thing existed.

File: tests/laser/evm_testsuite/evm_test.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
+from mythril.laser.ethereum.svm import LaserEVM
+from mythril.laser.ethereum.state import Account
+from mythril.disassembler.disassembly import Disassembly
+from mythril.laser.ethereum.transaction.concolic import execute_message_call
+from datetime import datetime
+from z3 import is_true, simplify
```
Comment (norhh) [inline tests/laser/evm_testsuite/evm_test.py]: these aren't being used right?

File: mythril/laser/ethereum/transaction/transaction_models.py (hunk 1 above)
Comment (JoranHonig) [reply mythril/laser/ethereum/transaction/transaction_models.py]: Creds to s0b0lev from discord for suggesting this. I didn't know either

File: tests/laser/evm_testsuite/evm_test.py (hunk 2 above)
Comment (JoranHonig) [reply tests/laser/evm_testsuite/evm_test.py]: Fixed

**Pull request merged**